contracts/EllipticCurve.sol

```diff
@@ -298,13 +296,10 @@ library EllipticCurve {
       mulmod(_x2, zs[0], _pp),
       mulmod(_y2, zs[1], _pp)
     ];
-    if (zs[0] == zs[2]) {
-      if (zs[1] != zs[3])
-        revert("Wrong data");
-      else {
-        revert("Use double instead");
-      }
-    }
+
+    // In case of zs[0] == zs[2] && zs[1] == zs[3], double function should be used
+    require(zs[0] != zs[2], "Invalid data");
+
```

Comment:
Now does not revert if zs[0] == zs[2] && zs[1] == z3[3]

Reply:
We grouped both reverts into a single require statement. :)